### PR TITLE
Add Initiator flag encoding/decoding in payload header

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
Currently, Initiator flag is set but not encoded in payload header, this field is now needed for ExchangeManager dispatching

 #### Summary of Changes
Add Initiator flag encoding/decoding in payload header

### CODE REVIEW

File: src/transport/raw/MessageHeader.cpp [hunk 1]
```diff
@@ -207,6 +208,17 @@ CHIP_ERROR PayloadHeader::Decode(Header::Flags flags, const uint8_t * const data
         mVendorId.ClearValue();
     }
 
+    mExchangeFlags.SetRaw(header);
```
Comment (andy31415) [inline src/transport/raw/MessageHeader.cpp:211]: Whad does the setInitiator do?

I believe the initiator flags should be based on mExchangeFlags, so checking mExchangeFlags to set mExchangeFlags seems like a noop.

File: src/transport/raw/MessageHeader.cpp [hunk 2]
```diff
@@ -266,9 +278,13 @@ CHIP_ERROR PayloadHeader::Encode(uint8_t * data, size_t size, uint16_t * encode_
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     uint8_t * p    = data;
+    uint8_t header = 0;
+
     VerifyOrExit(size >= EncodeSizeBytes(), err = CHIP_ERROR_INVALID_ARGUMENT);
 
-    Write8(p, mExchangeHeader);
+    header = header | mExchangeFlags.Raw();
```
Comment (andy31415) [inline src/transport/raw/MessageHeader.cpp:285]: why not uint8_t header = mExchangeFlags.Raw()

or even

`Write8(p, mExchangeFlags.Raw())`

?

Are there separate data bits in there? Should we even call it 'header' or should it be exchange flags?

File: src/transport/raw/MessageHeader.cpp (hunk 1 above)
Comment (yufengwangca) [reply src/transport/raw/MessageHeader.cpp:211]: Yes, mExchangeFlags.SetRaw(header); has already handled flag set, the setInitiator operation is not needed. Acked.

File: src/transport/raw/MessageHeader.cpp (hunk 2 above)
Comment (yufengwangca) [reply src/transport/raw/MessageHeader.cpp:276]: To align with  PacketHeader::Encode(), we don't need to assume the size of the mExchangeFlags.Raw() with this way.

Comment (yufengwangca) [reply src/transport/raw/MessageHeader.cpp]: Acked

File: src/transport/raw/MessageHeader.cpp [hunk 3]
```diff
@@ -266,9 +269,11 @@ CHIP_ERROR PayloadHeader::Encode(uint8_t * data, size_t size, uint16_t * encode_
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     uint8_t * p    = data;
+    uint8_t header = mExchangeFlags.Raw();
```
Comment (andy31415) [inline src/transport/raw/MessageHeader.cpp:272]: I would drop the extra stack variable and do `Write8(p, mExchangeFlags.Raw())`. Saves a level of indirection.

**Review by andy31415 — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Pull request merged**